include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -89,17 +89,29 @@ A_{3, 2}  A_{3, 3}  A_{3, 0}  A_{3, 1} ...   [phase 1] /
           if (isKDimInner) {
             const int numBanks = 32;
             const int bankBitWidth = 32;
+            const int SIMDWidth = 16;
 
             // number of inner dimension rows per one pattern repeat
-            int outerDimGranularity = mfmaEnc.getNonKDim();
             int typeBitWidth = eltTy.getIntOrFloatBitWidth();
             int innerDimLength = shape[order[0]];
             int elemsPerOneBanksRow = (numBanks * bankBitWidth) / typeBitWidth;
 
             int perPhase = std::max(1, elemsPerOneBanksRow / innerDimLength);
-            int maxPhase = outerDimGranularity / perPhase;
-            int vecSize = innerDimLength / maxPhase;
-            assert(vecSize > 0);
+            // Note: the following settings is customized to avoid
+            // **load** bank conflicts
+            //
+            // vecSize is set to k_base, which is the number of elements each
+            // workitem loads for one mfma instruction.
+            // For now, the k_base rules are as follows
+            // 1. All selected mfma instructions produce a single block
+            // 2. For f16 data type, 2 VGPRs are used for operand A --> k_base = 4
+            // 3. For non-f16 data types, 1 VGPR are used for operand A
+            //    k_base = 32 / elemTypeInBits
+            // 4. TODO: what about f64?
```

Comment:
Do we work for fp64 for now?

Reply:
I think this is a low-priority for now.
This also requires some changes in frontend(in upstream), if I remember it correctly.